server/rooms.ts

```diff
@@ -1559,6 +1552,24 @@ export class GameRoom extends BasicRoom {
 		this.sendUser(connection, '|init|battle\n|title|' + this.title + '\n' + this.getLogForUser(user));
 		if (this.game && this.game.onConnect) this.game.onConnect(user, connection);
 	}
+	/**
+	 * @param newTitle Specify this parameter if you need to change the room's title as well as its ID.
+	 */
+	renameBattle(newID: RoomID, newTitle?: string) {
+		if (!this.battle || this.tour) return;
+		const oldID = this.roomid;
+		this.roomid = newID;
+		if (newTitle) this.title = newTitle;
+		Rooms.rooms.delete(oldID);
+		Rooms.rooms.set(newID, this as GameRoom);
+
+		for (const user of Object.values(this.users)) {
+			user.moveConnections(oldID, newID);
+			user.send(`>${oldID}\n|noinit|rename|${newID}|${this.title}`);
+		}
+
+		return true;
+	}
```

Comment:
It worries me that this is a different function from the one for renaming rooms. That seems wrong to me.

Reply:
Is there a reason why this functionality can't be part of `rename`?

Reply:
it was discussed on discord in #sim-dev and iirc the consensus was that rensning without moving logs was different enough to warrant its own function.

Reply:
I disagree.

Even if there wasn't as much shared code as there is, it should at least be a function called by `rename`, rather than an entirely separate function.

But because there's so much shared code, I think it'd be better for it to be one function.

Reply:
Alright, fair.